CartoDB/cartodb
# Icon styling with categories

Closes #11019 

### Main changes
- Enabled style category ramps with icons.
- Style implemented using [Turbocarto ramps](https://github.com/CartoDB/turbo-carto#turbo-carto-ramps) in `marker-file` property thanks to @rochoa.
- SVG icon files filled with category ramp colors.

### How to test
0. Add the feature-flag `icon-styling` to the user you're going to use to test the feature.
1. Create a new map in CARTO Builder with a points geometry layer.
2. Go to Style tab, select no aggregation and in the fill input select filling by value.
3. Select a type string column. It should apply a default ramp without any image yet.
4. Add and remove icons from the categories, change colors. All those changes should be reflected in the map.
5. Use the Cartocss console and change the `marker-file` and `marker-fill` properties. Changes still should be reflected in the map.
6. Make other changes not related with categories (add layers, analysis, change fill to solid, etc). There shouldn't be any error or strange behaviour in the map.

<img width="1112" alt="screen shot 2017-01-05 at 10 25 51" src="https://cloud.githubusercontent.com/assets/2141690/21675257/7816b42a-d331-11e6-8c10-eea1b1093b40.png">
<img width="1113" alt="screen shot 2017-01-05 at 10 26 18" src="https://cloud.githubusercontent.com/assets/2141690/21675259/78226496-d331-11e6-88af-96d41df493c8.png">
<img width="1112" alt="screen shot 2017-01-05 at 10 26 32" src="https://cloud.githubusercontent.com/assets/2141690/21675258/78221432-d331-11e6-8e3c-ca2cd1750095.png">

Special thanks to @javierarce and @ivanmalagon who helped a lot developing this feature.
